common/archiver/README.md

```diff
@@ -0,0 +1,69 @@
+## What
+
+This README explains how to create new implementations for the History and Visibility Archiver Interface.
+
+## Steps
+
+1. Create a new directory in the `archiver` folder. The structure should looks like the following:
+
+```
+./common/archiver
+  - filestore/                  -- Filestore implementation 
+  - provider/
+      - provider.go             -- Provider of archiver instances
+  - yourImplementation/
+      - historyArchiver.go      -- HistoryArchiver implementation
+      - visibilityArchiver.go   -- VisibilityArchiver implementations
+```
+2. Each Archiver will be given a `BootstrapContainer` when it's created and the archiver implementation can (will) use them to read history, emit logs/metrics etc. Check `interface.go` for more details.
+
+3. Implement the Archiver interface. Both Archiver interface contains three methods:
+  
+```go
+func Archive(ctx context.Context, URI string, request *ArchiveRequest, opts ...ArchiveOption) error
+
+func Get(ctx context.Context, URI string, request *GetRequest) (*GetResponse, error)
+
+func ValidateURI(URI string) error
+```
+
+Each archiver implementation should define a format of URI, which is used to describe the type of implementation and the target location of `Archive()` or `Get()` operation. For example, the URI for filestore uses the format: `file:///path/to/dir`. The `ValidateURI()` method is responsible for checking if a given string is a valid URI.
+
+The `Archive()` method should archive workflow histories or visibility records described by the archive request to the location specified by the URI, while the `Get()` method is responsible for retrieving those archived data.
+
+Since the `Archive()` method will be called from different places (for example, timerQueueProcessor and worker), it may want to do different things. Therefore, a list of `ArchiveOption` will be passed to this method. These options will be applied to an `ArchiveFeatureCatalog`, and by checking the fields of that catalog, the `Archive()` method can figure out what to do. Right now, the only feature in the catalog is `ProgressManager` which can be used to record and load archive progress. More features can be added if needed. The correct way to use it is shown in the code sample below.
```

Comment:
I think its more instructive to say the "Archive method may be invoked differently by different callers. For examples some callers may automatically retry, while others only try once. Therefore, a list of ArchiveOptions..."

Reply:
Updated.